test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -611,6 +621,26 @@ foo
             }
         }
     }
+
+    Context "CustomPipeName startup tests" {
+
+        It "Should create pipe file if CustomPipeName is specified" {
+            $pipeName = [System.IO.Path]::GetRandomFileName()
+            $pipePath = Get-PipePath $pipeName
+
+            # The pipePath should be created by the time the -Command is executed.
+            pwsh -CustomPipeName $pipeName -Command "Test-Path '$pipePath'" | Should -BeTrue
+        }
+
+        It "Should throw if CustomPipeName is too long on Linux or macOS" -Skip:($IsWindows) {
+            $longPipeName = "DoggoipsumwaggywagssmolborkingdoggowithalongsnootforpatsdoingmeafrightenporgoYapperporgolongwatershoobcloudsbigolpupperlengthboy"
+
+            "`$pid" | & $powershell -CustomPipeName $longPipeName -c -
+            # 64 is the ExitCode for BadCommandLineParameter
+            $LASTEXITCODE | Should -Be 64
```

Comment:
> Will we see more user friendly message? Should we check it?

Not good to test for that because of Culture since the error message is in a resx.

Reply:
You can check for `FullyQualifiedErrorId` that is not localized.

Reply:
@adityapatwardhan in this case, I don't think you can because this is not an exception thrown, it's only an EXITCODE that is set along with some error string. It's pwsh.exe trying to start up... not some command that's failing to run.

an example:
```
.\pwsh.exe -custompipename
Cannot process the command because -CustomPipeName requires an argument that is a name of the pipe you want to use. Specify this argument and try again.
```